airflow/providers/google/cloud/operators/dataproc.py

```diff
@@ -943,9 +947,8 @@ def __init__(
 
         self.job_error_states = job_error_states if job_error_states is not None else {'ERROR'}
         self.impersonation_chain = impersonation_chain
-
         self.hook = DataprocHook(gcp_conn_id=gcp_conn_id, impersonation_chain=impersonation_chain)
```

Comment:
Hook should be initialized in execute method to avoid connection to metadata  server at dag loading.

Reply:
Thats a really good point, but we won't be able to get the project_id from the hook if the hook is not yet instantiated 🤔 We would also have to update the generate_job function in all of the operators which inherit from this one.